### DISCUSSION TITLE
Apple Intel macOS Ventura 13.1(M1): unable to install ruby 2.7.7, 3.1.3, 3.2.0, 3.0.5

### DISCUSSION THREAD

**Discussion by woolfiekim** [rbenv/ruby-build#2127] [Build failures] [answered]
Help me.... plz help me...

i tried install ruby 100times for create git blog by local.

but 

```shell
❯ rbenv install 2.7.7
To follow progress, use 'tail -f /var/folders/_q/0v9yhl5j1wz64t17fwvzljsw0000gn/T/ruby-build.20230116161632.64650.log' or pass --verbose
Downloading openssl-1.1.1s.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/c5ac01e760ee6ff0dab61d6b2bbd30146724d063eb322180c6f18a6f74e4b6aa
Installing openssl-1.1.1s...
Installed openssl-1.1.1s to /Users/kim_joohee/.rbenv/versions/2.7.7

Downloading ruby-2.7.7.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.7/ruby-2.7.7.tar.bz2
Installing ruby-2.7.7...

WARNING: ruby-2.7.7 is nearing its end of life.
It only receives critical security updates, no bug fixes.

ruby-build: using readline from homebrew
ruby-build: using gmp from homebrew

BUILD FAILED (macOS 13.1 using ruby-build 20221225)

Inspect or clean up the working tree at /var/folders/_q/0v9yhl5j1wz64t17fwvzljsw0000gn/T/ruby-build.20230116161632.64650.w87tlo
Results logged to /var/folders/_q/0v9yhl5j1wz64t17fwvzljsw0000gn/T/ruby-build.20230116161632.64650.log

Last 10 log lines:
compiling objspace_dump.c
compiling date_strftime.c
compiling date_strptime.c
installing default date_core libraries
installing default console libraries
linking shared-object objspace.bundle
linking shared-object io/console.bundle
linking shared-object nkf.bundle
linking shared-object date_core.bundle
make: *** [build-ext] Error 2
```

I tried many things.

1. 
```shell
$ brew install libyaml
$ brew update && brew upgrade ruby-build
```

2.  reinstall Xcode
3. update ventura
4. use docker
5. uninstall ruby
```shell
❯ rbenv uninstall 2.6.10p210
rbenv: version `2.6.10p210' not installed
❯ rbenv uninstall 2.6.1
rbenv: version `2.6.1' not installed
❯ rbenv uninstall 2.6.10
rbenv: version `2.6.10' not installed
```
but I still unable to install ruby. 

+ 
```shell
❯ ruby --version
ruby 2.6.10p210 (2022-04-12 revision 67958) [universal.arm64e-darwin22]
```

**Comment by elyxcode** [comment]
Hi, I had a problem similar to yours. I tried to install version 2.7.6. 3.2.0 failed to install them. So I did in the rbenv installation guide there is a link that this resource [Suggested build environment](https://github.com/rbenv/ruby-build/wiki#suggested-build-environment) in the mac os section executes the commands.

`brew install openssl@3 readline libyaml gmp
export RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@3)"`

 I used this because I wanted to install the 3. above but it is for the 2. too

also install rust
`brew install rust`

Then I tried to install 3.2.0 again and I succeeded.
My version of macOS is M1 Ventura 13.1

Try to see if it works for you.

**Reply by sandeep-kandra** [reply]
still not working for me , 
getting same error 
Your Ruby version is 2.6.10, but your Gemfile specified 2.7.6

**Reply by kushavi2509** [reply]
I tried with "rbenv", but I guess there is some issue with rbenv.
Try  removing rbenv > **CLOSE ALL TERMINAL** >then  install using "rvm"
\curl -sSL https://get.rvm  
 rvm install ${VERSION}

**Comment by mislav** [comment, marked as answer]
@woolfiekim Often people are better able to help with build errors if you include the full log output. By default, ruby-build does not print full output because that would be overwhelming for most usages. However, the output does contain the filename where the full log is written. You can paste that file as a GitHub Gist, edit your original question, and paste the link there.

**Reply by mislav** [reply]
In the meantime, the OP has solved their problem by installing Ruby 2.7 using Homebrew instead of with ruby-build, and since they haven't shared the build log, we are unable to find out what was the problem, and I'm going to lock this thread to reduce noise.

FWIW, I was able to install Ruby 2.7.7 on an M1 macOS 13.2 machine:
```sh
brew install openssl@1.1 libyaml gmp readline
RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1) --disable-install-doc" rbenv install 2.7.7
```

Note that Ruby 2.7 is becoming unsupported 3 days from now (end of March 2023). Please upgrade your apps to a newer Ruby.

**Comment by woolfiekim** [comment]
I solved this problem myself. 
How to solved...

I used 'brew' instead 'rbenv'

```shell
$ brew install ruby@2.7
$ echo 'export PATH="/usr/local/opt/ruby@2.7/bin:$PATH"'
$ bundle install
$ bundel exec jekyll serve
```

then I success!!! 

YESSSSSSSS!!!!!

**Reply by sandeep-kandra** [reply]
still not working for me , 
getting same error 
Your Ruby version is 2.6.10, but your Gemfile specified 2.7.6

**Comment by easyhooon** [comment]
I referred to the article on that link https://codecamper.me/blog/122/

If the ruby version does not change after executing the rbenv global command,
This requires setting environment variables.

Run the rbenv init and paste the output eval ~ line after the file that appears in the second line.

$ rbenv init
// Load rbenv automatically by appending
// the following to ~/.zshrc:

eval "$(rbenv init - zsh)"

if file that appears in the second line is ~/.zshrc
$ open ~/.zshrc 

and paste eval "$(rbenv init - zsh)"

If you run the terminal again after pasting the above, you can see that the ruby version is applied well.

$ ruby -v
ruby 2.7.7p221 (2022-11-24 revision 168ec2b1e5) [arm64-darwin22]
